tests/test_bgp_af.rb

```diff
@@ -457,50 +671,73 @@ def test_dampening
     # Set and verify 'dampening' with defaults #
     ############################################
     vrf = 'orange'
+    if platform == :ios_xr
+      config_ios_xr_dependencies(asn)
+      dampening_vrf_xr_neg_test
+      vrf = 'default'
+    end
     af = %w(ipv4 unicast)
     bgp_af = RouterBgpAF.new(asn, vrf, af)
 
     # Test no dampening configured
     assert_nil(bgp_af.dampening)
 
-    pattern = /^ *dampening$/
+    if platform == :nexus
+      pattern = /^ *dampening$/
+    elsif platform == :ios_xr
+      pattern = /^ *bgp dampening$/
+    end
 
     bgp_af.dampening = []
 
     # Check property got set
-    af_string = get_bgp_af_cfg(asn, vrf, af)
+    if platform == :nexus
+      af_string = get_bgp_af_cfg(asn, vrf, af)
+    elsif platform == :ios_xr
+      af_string = @device.cmd("show run router bgp #{asn} " \
+                              'address-family ipv4 unicast bgp dampening')
+    end
 
     assert_match(pattern, af_string,
                  "Error: 'dampening' is not configured and should be")
 
-    # Check properties got set
-    af_string = get_bgp_af_dampening_params(asn, vrf, af)
+    if platform == :nexus
 
-    pattern_params1 = 'Half-life time                 : 15 mins'
-    pattern_params2 = 'Suppress penalty               : 2000'
-    pattern_params3 = 'Reuse penalty                  : 750'
-    pattern_params4 = 'Max suppress time              : 45 mins'
-    pattern_params5 = 'Max suppress penalty           : 6000'
+      # Check properties got set
+      af_string = get_bgp_af_dampening_params(asn, vrf, af)
 
-    error = ("Error: 'dampening' properties are incorrect")
+      pattern_params1 = 'Half-life time                 : 15 mins'
+      pattern_params2 = 'Suppress penalty               : 2000'
+      pattern_params3 = 'Reuse penalty                  : 750'
+      pattern_params4 = 'Max suppress time              : 45 mins'
+      pattern_params5 = 'Max suppress penalty           : 6000'
 
-    assert_match(pattern_params1, af_string, error)
-    assert_match(pattern_params2, af_string, error)
-    assert_match(pattern_params3, af_string, error)
-    assert_match(pattern_params4, af_string, error)
-    assert_match(pattern_params5, af_string, error)
+      error = ("Error: 'dampening' properties are incorrect")
 
-    # Check getter
-    assert_empty(bgp_af.dampening, 'Error: dampening is configured and ' \
-                 'should not be')
+      assert_match(pattern_params1, af_string, error)
+      assert_match(pattern_params2, af_string, error)
+      assert_match(pattern_params3, af_string, error)
+      assert_match(pattern_params4, af_string, error)
+      assert_match(pattern_params5, af_string, error)
+      # Check getter
+      assert_empty(bgp_af.dampening, 'Error: dampening is configured and ' \
+                   'should not be')
+    end
+    # "show bgp dampening param not supported on IOS XR"
```

Comment:
Seems like we should be testing the getters here for both NX and XR to make sure they return the correct default values.